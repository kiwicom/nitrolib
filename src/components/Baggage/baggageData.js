// @flow
export default {
  definitions: {
    handBag: [
      {
        note: null,
        price: {
          currency: "EUR",
          amount: 0,
        },
        currency: "EUR",
        category: "personalItem",
        restrictions: {
          weight: 5,
          height: 20,
          width: 20,
          length: 20,
          dimensions_sum: null,
        },
      },
      {
        note: null,
        price: {
          currency: "EUR",
          amount: 0,
        },
        currency: "EUR",
        category: "cabinBag",
        restrictions: {
          weight: 10,
          height: 35,
          width: 20,
          length: 45,
          dimensions_sum: null,
        },
      },
    ],
    holdBag: [
      {
        note: null,
        price: {
          currency: "EUR",
          amount: 10,
        },
        currency: "EUR",
        category: "holdBag",
        restrictions: {
          weight: 10,
          height: 52,
          width: 26,
          length: 78,
          dimensions_sum: 156,
        },
      },
      {
        note: null,
        price: {
          currency: "EUR",
          amount: 25,
        },
        currency: "EUR",
        category: "holdBag",
        restrictions: {
          weight: 15,
          height: 52,
          width: 26,
          length: 78,
          dimensions_sum: 156,
        },
      },
      {
        note: null,
        price: {
          currency: "EUR",
          amount: 30,
        },
        currency: "EUR",
        category: "holdBag",
        restrictions: {
          weight: 25,
          height: 52,
          width: 26,
          length: 78,
          dimensions_sum: 156,
        },
      },
    ],
  },
  combinations: {
    adult: {
      handBag: [
        {
          category: "handBag",
          group: "adult",
          combination: [0],
          price: {
            currency: "EUR",
            amount: 0,
          },
          originalIndex: 0,
          allowanceGraph: {
            add: {},
            remove: {},
          },
        },
        {
          category: "handBag",
          group: "adult",
          combination: [1],
          price: {
            currency: "EUR",
            amount: 15,
          },
          originalIndex: 1,
          allowanceGraph: {
            add: {},
            remove: {},
          },
        },
        {
          category: "handBag",
          group: "adult",
          combination: [0, 1],
          price: {
            currency: "EUR",
            amount: 20,
          },
          originalIndex: 2,
          allowanceGraph: {
            add: {},
            remove: {},
          },
        },
        {
          category: "handBag",
          group: "adult",
          combination: [0, 1, 1],
          price: {
            currency: "EUR",
            amount: 30,
          },
          originalIndex: 3,
          allowanceGraph: {
            add: {},
            remove: {},
          },
        },
        {
          category: "handBag",
          group: "adult",
          combination: [0, 1, 1, 1],
          price: {
            currency: "EUR",
            amount: 40,
          },
          originalIndex: 4,
          allowanceGraph: {
            add: {},
            remove: {},
          },
        },
      ],
      holdBag: [
        {
          category: "holdBag",
          group: "adult",
          combination: [],
          price: {
            currency: "EUR",
            amount: 0,
          },
          originalIndex: 5,
          allowanceGraph: {
            add: {
              "0": 1,
              "1": 2,
              "2": 3,
            },
            remove: {},
          },
        },
        {
          category: "holdBag",
          group: "adult",
          combination: [0],
          price: {
            currency: "EUR",
            amount: 10,
          },
          originalIndex: 6,
          allowanceGraph: {
            add: {},
            remove: {
              "0": 0,
            },
          },
        },
        {
          category: "holdBag",
          group: "adult",
          combination: [1],
          price: {
            currency: "EUR",
            amount: 25,
          },
          originalIndex: 7,
          allowanceGraph: {
            add: {},
            remove: {
              "1": 0,
            },
          },
        },
        {
          category: "holdBag",
          group: "adult",
          combination: [2],
          price: {
            currency: "EUR",
            amount: 30,
          },
          originalIndex: 8,
          allowanceGraph: {
            add: {},
            remove: {
              "2": 0,
            },
          },
        },
        {
          category: "holdBag",
          group: "adult",
          combination: [1, 1, 2],
          price: {
            currency: "EUR",
            amount: 50,
          },
          originalIndex: 9,
          allowanceGraph: {
            add: {},
            remove: {
              "2": 0,
            },
          },
        },
        {
          category: "holdBag",
          group: "adult",
          combination: [1, 1, 2, 2],
          price: {
            currency: "EUR",
            amount: 60,
          },
          originalIndex: 10,
          allowanceGraph: {
            add: {},
            remove: {
              "2": 0,
            },
          },
        },
      ],
    },
    child: {
      handBag: [
        {
          category: "handBag",
          group: "adult",
          combination: [0, 1],
          price: {
            currency: "EUR",
            amount: 0,
          },
          originalIndex: 0,
          allowanceGraph: {
            add: {},
            remove: {},
          },
        },
      ],
      holdBag: [
        {
          category: "holdBag",
          group: "adult",
          combination: [],
          price: {
            currency: "EUR",
            amount: 0,
          },
          originalIndex: 1,
          allowanceGraph: {
            add: {
              "0": 1,
              "1": 2,
              "2": 3,
            },
            remove: {},
          },
        },
        {
          category: "holdBag",
          group: "adult",
          combination: [0],
          price: {
            currency: "EUR",
            amount: 0,
          },
          originalIndex: 2,
          allowanceGraph: {
            add: {},
            remove: {
              "0": 0,
            },
          },
        },
        {
          category: "holdBag",
          group: "adult",
          combination: [1],
          price: {
            currency: "EUR",
            amount: 25,
          },
          originalIndex: 3,
          allowanceGraph: {
            add: {},
            remove: {
              "1": 0,
            },
          },
        },
        {
          category: "holdBag",
          group: "adult",
          combination: [2],
          price: {
            currency: "EUR",
            amount: 30,
          },
          originalIndex: 4,
          allowanceGraph: {
            add: {},
            remove: {
              "2": 0,
            },
          },
        },
      ],
    },
    infant: {
      handBag: [
        {
          originalIndex: null,
          allowanceGraph: {
            add: {},
            remove: {},
          },
          category: "handBag",
          combination: [],
          group: "infant",
          price: {
            currency: "EUR",
            amount: 0,
          },
        },
      ],
      holdBag: [
        {
          originalIndex: null,
          allowanceGraph: {
            add: {},
            remove: {},
          },
          category: "holdBag",
          combination: [],
          group: "infant",
          price: {
            currency: "EUR",
            amount: 0,
          },
        },
      ],
    },
  },
};
