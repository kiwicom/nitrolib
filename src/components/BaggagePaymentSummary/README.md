# BaggagePaymentSummary

- renders baggage payment summary
- renders baggages per passenger with price per passenger
- renders summary price for all baggages in itinerary

**Example:**

```js
<BaggagePaymentSummary
  passengers={[
    {
      paxId: 1,
      firstName: "Vaclav",
      lastName: "Havel",
      baggage: {
        holdBag: 1,
        handBag: 1
      }
    }
  ]}
  baggage={baggageData}
/>
```
