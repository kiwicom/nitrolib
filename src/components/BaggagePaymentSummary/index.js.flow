// @flow strict
import type { BaggageType } from "../../records/Baggage";

type Passenger = {|
  paxId: number,
  firstName: string,
  lastName: string,
  baggage: {
    holdBag: number,
    handBag: number,
  },
|};

type Props = {|
  passengers: Passenger[],
  baggage: BaggageType,
|};

declare export default React.ComponentType<Props>;
