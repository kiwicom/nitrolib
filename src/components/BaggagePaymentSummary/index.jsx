// @flow strict
import * as React from "react";
import Card from "@kiwicom/orbit-components/lib/Card";
import CardSection from "@kiwicom/orbit-components/lib/Card/CardSection";
import Separator from "@kiwicom/orbit-components/lib/Separator";

import PassengerBaggages from "./components/PassengerBaggages";
import TotalPayment from "./components/TotalPayment";
import type { BaggageType } from "../../records/Baggage";
import { getTotalPrice } from "../../services/baggage/utils";
import getDefinitions from "./services/getDefinitions";

type Passenger = {
  paxId: number,
  firstName: string,
  lastName: string,
  baggage: {
    holdBag: number, // index of baggage combination
    handBag: number, // index of baggage combination
  },
};

type Props = {
  passengers: Passenger[],
  baggage: BaggageType,
};

const BaggagePaymentSummary = ({ baggage, passengers }: Props) => {
  const { combinations, definitions } = baggage;

  const passengersWithBagDefinitions = passengers.map(passenger => ({
    paxId: passenger.paxId,
    firstName: passenger.firstName,
    lastName: passenger.lastName,
    baggage: {
      handBag: getDefinitions(
        definitions.handBag,
        combinations.handBag[passenger.baggage.handBag].indices,
      ),
      holdBag: getDefinitions(
        definitions.holdBag,
        combinations.holdBag[passenger.baggage.holdBag].indices,
      ),
    },
    price:
      combinations.handBag[passenger.baggage.handBag].price.amount +
      combinations.holdBag[passenger.baggage.holdBag].price.amount,
  }));

  const passengersCombinationsIndices = {
    handBag: passengers.map(passenger => passenger.baggage.handBag),
    holdBag: passengers.map(passenger => passenger.baggage.holdBag),
  };

  return (
    <Card dataTest="BaggagePaymentSummary">
      <CardSection>
        {passengersWithBagDefinitions.map(passenger => (
          <PassengerBaggages key={passenger.paxId} {...passenger} />
        ))}
        <Separator spaceAfter="large" />
        <TotalPayment
          totalPrice={getTotalPrice({
            combinationIndices: passengersCombinationsIndices,
            combinations,
          })}
        />
      </CardSection>
    </Card>
  );
};
export default BaggagePaymentSummary;
