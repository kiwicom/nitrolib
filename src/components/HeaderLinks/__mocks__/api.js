// @flow strict

// TODO move to HeaderLinks/records/HeaderLink.js
export const props = {
  languageId: "de",
  currencyId: "GBP",
  searchForm: {
    mode: "oneWay",
    destination: {
      type: "2",
      name: "Warsaw",
    },
    checkIn: new Date(),
    checkOut: null,
    adults: 1,
    children: 0,
  },
  splitster: {
    HEADER_LINKS_ROOMS_PROVIDER_0: "roomsKiwiCode",
  },
  context: "kiwicom",
};

export const response = {
  splitster: {
    HEADER_LINKS_ROOMS_PROVIDER_0: "roomsKiwiCode",
  },
  items: [
    {
      id: "travel",
      newWindow: false,
      translation: "search.service.travel_anywhere",
      url:
        "https://red-cougar.kiwi.com/nav-bar-link?u=TRAVEL&r=search&lang=us&payload=eyJjYXRlZ29yeSI6Ik5hdkJhciIsInN1YkNhdGVnb3J5IjoiTGluayIsImFjdGlvbiI6IkNsaWNrIiwiZGV0YWlsIjoia2l3aS5jb20gLSB0cmF2ZWwifQ==",
    },
    {
      id: "rooms",
      newWindow: true,
      translation: "search.service.rooms",
      url:
        "https://red-cougar.kiwi.com/nav-bar-link?u=ROOMS_KIWI&query=aW5kZXguaHRtbD9zZWxlY3RlZF9jdXJyZW5jeT1FVVImaGVhZGVybGlua3M=&payload=eyJjYXRlZ29yeSI6Ik5hdkJhciIsInN1YkNhdGVnb3J5IjoiTGluayIsImFjdGlvbiI6IkNsaWNrIiwiZGV0YWlsIjoiUk9PTVNfS0lXSSAtIHJvb21zIn0=",
    },
    {
      id: "cars",
      newWindow: true,
      translation: "search.service.cars",
      url:
        "https://red-cougar.kiwi.com/nav-bar-link?u=CARS&query=cHJlZmxhbmc9dXMmYWRwbGF0PWhlYWRlcmxpbmtz&r=search&lang=us&payload=eyJjYXRlZ29yeSI6Ik5hdkJhciIsInN1YkNhdGVnb3J5IjoiTGluayIsImFjdGlvbiI6IkNsaWNrIiwiZGV0YWlsIjoia2l3aS5jb20gLSBjYXJzIn0=",
    },
    {
      id: "logitravel",
      newWindow: true,
      translation: "search.service.holidays",
      url:
        "https://red-cougar.kiwi.com/nav-bar-link?u=LOGITRAVEL&query=YWNudGI9RFAmYmZfc3Vic291cmNlPS0tLS0tVEwwUzEwUlIwMSZ1dG1fbWVkaXVtPXdoaXRlbGFiZWwmdXRtX3NvdXJjZT1raXdp&r=search&lang=us&payload=eyJjYXRlZ29yeSI6Ik5hdkJhciIsInN1YkNhdGVnb3J5IjoiTGluayIsImFjdGlvbiI6IkNsaWNrIiwiZGV0YWlsIjoibGFzdG1pbnV0ZS5jb20gLSBsb2dpdHJhdmVsIn0=",
    },
  ],
};
