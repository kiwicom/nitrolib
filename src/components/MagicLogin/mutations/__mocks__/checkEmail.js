// @noflow

const checkEmail = jest.fn((environment, email) => {
  if (email === "error@example.com") {
    return Promise.reject(new Error("Network issue"));
  }

  const result = {
    hasKiwiAccount: false,
    hasFacebook: false,
    hasGoogle: false,
    hasBooking: false,
  };

  if (["withBooking@example.com", "withBookingError@example.com"].includes(email)) {
    return Promise.resolve({
      checkEmail: {
        result: { ...result, hasBooking: true },
      },
    });
  }

  if (email === "withFacebook@example.com") {
    return Promise.resolve({
      checkEmail: {
        result: { ...result, hasFacebook: true, hasKiwiAccount: true, hasBooking: true },
      },
    });
  }

  return Promise.resolve({
    checkEmail: {
      result,
    },
  });
});

export default checkEmail;
