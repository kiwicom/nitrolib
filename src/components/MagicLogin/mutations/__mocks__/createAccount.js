// @noflow

const createAccount = jest.fn((environment, brand, credentials) => {
  if (credentials.email === "error@example.com") {
    return Promise.resolve({
      createAccount: {
        success: false,
        error: "INVALID_EMAIL",
      },
    });
  }

  return Promise.resolve({
    createAccount: {
      success: true,
      error: null,
    },
  });
});

export default createAccount;
