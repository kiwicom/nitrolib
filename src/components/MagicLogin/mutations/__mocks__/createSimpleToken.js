// @noflow

const createSimpleToken = jest.fn((environment, variables) => {
  if (variables.input.email === "nobooking@example.com") {
    return Promise.resolve({
      createSimpleToken: {
        code: "NOT_FOUND",
      },
    });
  }

  return Promise.resolve({
    createSimpleToken: {
      token: "abcd-1234-aaaa-1111",
    },
  });
});

export default createSimpleToken;
