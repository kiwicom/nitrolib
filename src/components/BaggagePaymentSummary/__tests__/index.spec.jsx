// @flow strict
import * as React from "react";
import { mount } from "enzyme";

import BaggagePaymentSummary from "..";

import { baggageData } from "../../../records/__mocks__/baggageData";

const passengers = [
  {
    paxId: 1,
    firstName: "Barrack",
    lastName: "Obama",
    baggage: {
      holdBag: 1, // index of baggage combination
      handBag: 1, // index of baggage combination
    },
  },
  {
    paxId: 2,
    firstName: "Donald",
    lastName: "Trump",
    baggage: {
      holdBag: 2, // index of baggage combination
      handBag: 2, // index of baggage combination
    },
  },
];

const props = {
  passengers,
  baggage: baggageData,
};

describe("#BaggagePaymentSummary", () => {
  test("renders", () => {
    const wrapper = mount(<BaggagePaymentSummary {...props} />);
    expect(wrapper.find("[data-test='BaggagePaymentSummary']").exists()).toBe(true);
  });
});
