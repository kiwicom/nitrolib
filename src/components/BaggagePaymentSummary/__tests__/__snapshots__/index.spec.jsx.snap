// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#BaggagePaymentSummary renders 1`] = `
<Card>
  <CardSection>
    <PassengerBaggages
      baggage={
        Object {
          "handBag": Array [
            Object {
              "amount": 1,
              "category": "personalItem",
              "conditions": Object {
                "passengerGroups": Array [
                  "adult",
                ],
              },
              "restrictions": Object {
                "dimensionsSum": null,
                "height": 20,
                "length": 20,
                "weight": 5,
                "width": 20,
              },
            },
          ],
          "holdBag": Array [
            Object {
              "amount": 1,
              "category": "holdBag",
              "conditions": Object {
                "passengerGroups": Array [
                  "adult",
                ],
              },
              "restrictions": Object {
                "dimensionsSum": 156,
                "height": 52,
                "length": 78,
                "weight": 10,
                "width": 26,
              },
            },
          ],
        }
      }
      firstName="Barrack"
      id={1}
      key="1"
      lastName="Obama"
      price={10.11}
    />
    <PassengerBaggages
      baggage={
        Object {
          "handBag": Array [
            Object {
              "amount": 1,
              "category": "cabinBag",
              "conditions": Object {
                "passengerGroups": Array [
                  "adult",
                ],
              },
              "restrictions": Object {
                "dimensionsSum": null,
                "height": 35,
                "length": 45,
                "weight": 10,
                "width": 20,
              },
            },
          ],
          "holdBag": Array [
            Object {
              "amount": 1,
              "category": "holdBag",
              "conditions": Object {
                "passengerGroups": Array [
                  "adult",
                ],
              },
              "restrictions": Object {
                "dimensionsSum": 156,
                "height": 52,
                "length": 78,
                "weight": 15,
                "width": 26,
              },
            },
          ],
        }
      }
      firstName="Donald"
      id={1}
      key="1"
      lastName="Trump"
      price={30.33}
    />
    <Separator
      spaceAfter="large"
    />
    <TotalPayment
      totalPrice={40.44}
    />
  </CardSection>
</Card>
`;
